odl/phantom/geometric_phantoms.py

```diff
@@ -0,0 +1,189 @@
+
+        slice1[nn] = slice(start, stop)
+
+    phan[slice1] = 1
+
+    return discr_space.element(phan)
+
+
+def indicate_proj_axis(discr_space, scale_structures=0.5):
+    """Phantom indicating along which axis it is projected.
+
+    The number (n) of rectangles in a parallel-beam projection along a main
+    axis (0, 1, or 2) indicates the projection to be along the (n-1)the
+    dimension.
+
+    Parameters
+    ----------
+    discr_space : `DiscretizedSpace`
+        Discretized space in which the phantom is supposed to be created
+    scale_structures : positive `float` in (0, 1]
+        Scales objects (cube, cuboids)
+
+    Returns
+    -------
+    phantom : `LinearSpaceVector`
+        Returns an element in ``discr_space``
+
+    Examples
+    --------
+    >>> import odl
+    >>> space = odl.uniform_discr([0] * 3, [1] * 3, [8, 8, 8])
+    >>> phan = indicate_proj_axis(space).asarray()
+    >>> print(np.sum(phan, 0))
+    [[ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  3.  3.  0.  0.  0.]
+     [ 0.  0.  0.  3.  3.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]]
+    >>> print(np.sum(phan, 1))
+    [[ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  2.  2.  0.  0.  0.]
+     [ 0.  0.  0.  2.  2.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  1.  1.  0.  0.  0.]
+     [ 0.  0.  0.  1.  1.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]]
+    >>> print(np.sum(phan, 2))
+    [[ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  2.  2.  0.  0.  0.]
+     [ 0.  0.  0.  2.  2.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  2.  0.  0.  0.]
+     [ 0.  0.  0.  2.  0.  0.  0.  0.]
+     [ 0.  0.  0.  0.  0.  0.  0.  0.]]
+    """
+    if not 0 < scale_structures <= 1:
+        raise ValueError('`scale_structures` ({}) is not in (0, 1]'
+                         ''.format(scale_structures))
+
+    assert discr_space.ndim == 3
+
+    shape = discr_space.shape
+    phan = np.zeros(shape)
+    shape = np.array(shape) - 1
+    cen = np.round(0.5 * shape)
+    dx = np.floor(scale_structures * 0.25 * shape)
+    dx[dx == 0] = 1
+
+    # cube of size 2 * dx
+    x0 = (cen - 3 * dx)[0]
+    x, y, z = cen - 1 * dx
+    phan[x0:x, y:-y, z:-z] = 1
+
+    # 1st cuboid of size (dx[0], dx[1], 2 * dx[2])
+    x0 = (cen + 1 * dx)[1]
+    x1 = (cen + 2 * dx)[1]
+    y0 = cen[1]
+    z = (cen - dx)[2]
+    phan[x0:x1, y0:-y, z:-z] = 1
+
+    # 2nd cuboid of (dx[0], dx[1], 2 * dx[2]) touching the first diagonally
+    # at a long edge
+    x0 = (cen + 2 * dx)[1]
+    x1 = (cen + 3 * dx)[1]
+    y1 = cen[1]
+    z = (cen - dx)[2]
+    phan[x0:x1, y:y1, z:-z] = 1
+
+    return discr_space.element(phan)
```

Comment:
@adler-j I know it's not your code, but I think it deserves some improvement anyway.